### PR TITLE
feat(stacktrace-link): Add RepositoryProjectPathConfig table

### PR DESCRIPTION
**Pull request by MeredithAnya**
**Context:**
Ecosystem is working on making stack trace linking a core piece of the repository provider integrations (GH, GitLab, etc). This means taking someone from a file in the Sentry stack trace to a file in their source code. There are challenges with finding out exactly where to link someone:

1. Integrations and repositories are on the **organization** level.
    * This makes it hard to go from event stack trace in a project to the correct repository if you have many repos that are connected to multiple projects. 
2. The file path in the stack trace may not be formatting quite right to link correctly to the source code - some additional formatting may be needed on a case to case basis.
3. The code may have changed since the error in Sentry was captured.
    * We can try to find the latest commit sha in order to redirect the user to the correct version of the code, but if we can't we may need a default branch to fall back on.

This why we've decided to have a configuration required, the configuration may change over time as we iterate, so the explanation in this PR description is may be outdated later.

**Configuration Fields:**

`project`, `repository`

* I think the `project` and `repository` are self explanatory, the `repository` has the integration_id and ~we can get an organization_integration from the project.organization and the integration.~

`organization_integration`

* We actually need this to be able to get the configurations easily when we load the integration configuration page 

`input_path`, `output_path`

* If needed, the `input_path` is the part of the stack trace file path that needs to be replaced with a different value, which is the `output_path`. If the file doesn't need re-formatting, both the input and output paths can be empty strings.

`default_branch`

* If we can't find the version of the code for the file path, we may need a default branch. Some providers may handle this, but others may still require that default be provided in the url.

### CODE REVIEW

File: src/sentry/models/integration.py [hunk 1]
```diff
@@ -48,6 +48,21 @@ class Meta:
         unique_together = (("organization_integration_id", "external_id"),)
 
 
+class RepositoryProjectPathConfig(DefaultFieldsModel):
+    __core__ = False
+
+    repository = FlexibleForeignKey("sentry.Repository")
+    project = FlexibleForeignKey("sentry.Project", db_constraint=False)
```
Comment (MeredithAnya) [inline src/sentry/models/integration.py:55]: @wedamija I wasn't sure if I should have added `db_index=False` here if the unique constraint will also create an index.

Comment (wedamija) [reply src/sentry/models/integration.py:55]: You can, it's probably fine either way. That's really just an optimisation, and this table probably won't be large enough or high volume enough for it to be important.

File: src/sentry/models/integration.py [hunk 2]
```diff
@@ -48,6 +48,21 @@ class Meta:
         unique_together = (("organization_integration_id", "external_id"),)
 
 
+class RepositoryProjectPathConfig(DefaultFieldsModel):
+    __core__ = False
+
+    repository = FlexibleForeignKey("sentry.Repository")
```
Comment (scefali) [inline src/sentry/models/integration.py:54]: @MeredithAnya should we have `db_constraint=False` here as well?

Comment (wedamija) [reply src/sentry/models/integration.py:54]: It's probably fine, I don't think repository is very high volume. Can't check at the moment :(

**Review by wedamija — APPROVED** [review]
Migration looks good to me

File: src/sentry/migrations/0112_add_repositoryprojectpathconfig.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+    operations = [
+        migrations.CreateModel(
+            name='RepositoryProjectPathConfig',
+            fields=[
+                ('id', sentry.db.models.fields.bounded.BoundedBigAutoField(primary_key=True, serialize=False)),
+                ('date_updated', models.DateTimeField(default=django.utils.timezone.now)),
+                ('date_added', models.DateTimeField(default=django.utils.timezone.now, null=True)),
+                ('input_path', models.TextField()),
+                ('output_path', models.TextField()),
```
Comment (markstory) [inline src/sentry/migrations/0112_add_repositoryprojectpathconfig.py:43]: stack_path and repository_path might be clearer names for how these columns would be used.

Comment (MeredithAnya) [reply src/sentry/migrations/0112_add_repositoryprojectpathconfig.py:43]: @scefali what are your thoughts? I like these names

Comment (scefali) [reply src/sentry/migrations/0112_add_repositoryprojectpathconfig.py:43]: @MeredithAnya maybe `base_stack_path` and `base_repository_path` might make more sense

Comment (MeredithAnya) [reply src/sentry/migrations/0112_add_repositoryprojectpathconfig.py:43]: what about `stack_root`, `source_root`?

Comment (scefali) [reply src/sentry/migrations/0112_add_repositoryprojectpathconfig.py:43]: @MeredithAnya sounds good to me

Comment (markstory) [reply src/sentry/migrations/0112_add_repositoryprojectpathconfig.py]: stack_root an source_root are 🌈

**Pull request merged**